terraform/projects/app-ecs-albs/main.tf

```diff
@@ -129,19 +129,127 @@ resource "aws_lb_listener" "monitoring_external_listener" {
   }
 }
 
+resource "aws_lb" "alertmanager_external_alb" {
+  name               = "${var.stack_name}-alertmanager"
+  internal           = true
+  load_balancer_type = "application"
+  security_groups    = ["${data.terraform_remote_state.infra_security_groups.aletmanager_external_sg_id}"]
+
+  subnets = [
+    "${element(data.terraform_remote_state.infra_networking.public_subnets, 0)}",
+    "${element(data.terraform_remote_state.infra_networking.public_subnets, 1)}",
+    "${element(data.terraform_remote_state.infra_networking.public_subnets, 2)}",
+  ]
+
+  tags = "${merge(
+    local.default_tags,
+    var.additional_tags,
+    map("Stackname", "${var.stack_name}"),
+    map("Name", "${var.stack_name}-ecs-monitoring")
+  )}"
+}
+
+resource "aws_lb_listener" "alertmanager_listener" {
+  load_balancer_arn = "${aws_lb.alertmanager_external_alb.arn}"
+  port              = "80"
+  protocol          = "HTTP"
+
+  default_action {
+    target_group_arn = "${aws_lb_target_group.alertmanager_endpoint.arn}"
+    type             = "forward"
+  }
+}
+
+resource "aws_lb_target_group" "alertmanager_endpoint" {
+  name     = "${var.stack_name}-alertmanager"
+  port     = "9093"
+  protocol = "HTTP"
+  vpc_id   = "${data.terraform_remote_state.infra_networking.vpc_id}"
+
+  health_check {
+    interval            = "10"
+    path                = "/"
+    matcher             = "200"
+    protocol            = "HTTP"
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+    timeout             = "5"
+  }
+}
+
+
+resource "aws_lb_listener" "paas_proxy_listener" {
+  load_balancer_arn = "${aws_lb.alertmanager_external_alb.arn}"
```

Comment:
why did you make this part of the alertmanager ALB? It's a bit weird as they don't have much to do with each other

Reply:
This was inline with digram so basically they can share an ALB as long we as we correctly define the forwarding rules  They share one internal ALB in the AZ. They are currently only routed based on port number. PaaS proxy is on port 8080 & Alertmanager is on 80.